src/Riok.Mapperly/Helpers/SymbolExtensions.cs

```diff
@@ -79,6 +79,49 @@ internal static IEnumerable<IPropertySymbol> GetAllAccessibleProperties(this ITy
         return genericIntf != null;
     }
 
+    internal static bool ImplementsGeneric(
+        this ITypeSymbol t,
+        INamedTypeSymbol genericInterfaceSymbol,
+        string symbolName,
+        [NotNullWhen(true)] out INamedTypeSymbol? genericIntf,
+        out bool isExplicit)
+    {
+        if (SymbolEqualityComparer.Default.Equals(t.OriginalDefinition, genericInterfaceSymbol))
+        {
+            genericIntf = (INamedTypeSymbol)t;
+            isExplicit = true;
+            return true;
+        }
+
+        genericIntf = t.AllInterfaces.FirstOrDefault(x => x.IsGenericType && SymbolEqualityComparer.Default.Equals(x.OriginalDefinition, genericInterfaceSymbol));
+
+        if (genericIntf == null)
+        {
+            isExplicit = false;
+            return false;
+        }
+
+        var interfaceSymbol = genericIntf.GetMembers(symbolName).Single();
```

Comment:
TBD: maybe switching to `First` instead of `Single` will be more resilient if a future release adds a method overload for a given method.